File: ai-robot/src/main/java/za/co/sww/rwars/airobot/model/LaserResponse.java
package za.co.sww.rwars.airobot.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record LaserResponse(
        boolean hit,
        String hitRobotId,
        String hitRobotName,
        int damageDealt,
        int range,
        String direction,
        List<Position> laserPath,
        Position hitPosition,
        String blockedBy
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Position(int x, int y) {}
}

File: robo-demo/src/main/kotlin/za/co/sww/rwars/robodemo/model/LaserResponse.kt
package za.co.sww.rwars.robodemo.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Response object for laser firing operations.
 */
data class LaserResponse(
    @JsonProperty("hit") val hit: Boolean,
    @JsonProperty("hitRobotId") val hitRobotId: String? = null,
    @JsonProperty("hitRobotName") val hitRobotName: String? = null,
    @JsonProperty("damageDealt") val damageDealt: Int = 0,
    @JsonProperty("range") val range: Int,
    @JsonProperty("direction") val direction: String,
    @JsonProperty("laserPath") val laserPath: List<Position> = emptyList(),
    @JsonProperty("hitPosition") val hitPosition: Position? = null,
    @JsonProperty("blockedBy") val blockedBy: String? = null,
) {
    /**
     * Position coordinates.
     */
    data class Position(
        @JsonProperty("x") val x: Int,
        @JsonProperty("y") val y: Int,
    )
}
